dts/arm/yaml/atmel,sam-uart.yaml

```diff
@@ -0,0 +1,30 @@
+---
+title: SAM Family UART
+id: atmel,sam-uart
+version: 0.1
+
+description: >
+    This binding gives a base representation of the SAM UART
+
+inherits:
+  - !include uart.yaml
+
+properties:
+  - compatible:
+      type: string
+      category: required
+      description: compatible strings
+      constraint: "atmel,sam-uart"
+
+  - reg:
```

Comment:
We should still have these in the device tree as we move to more code generation in Zephyr.  The plan is that a lot of the device data structs that we build up should eventually get generated.